### PR TITLE
Updated client, shared, deprecated, dx, gui

### PR DESCRIPTION
**Pull request by Loki214**
These were automatically scraped from the wiki with a little app my friend built. I'm sorry if there are any errors but doing this manually was taking forever lol

### CODE REVIEW

File: src/defs/deprecated.ts [hunk 1]
```diff
@@ -37,1035 +37,1035 @@ DeprecatedDefinitions.push(tmpDef);
...
\ No newline at end of file
+tmpDef.description = deprecatedText + "Please use xmlFindChild";
+tmpDef.returnType = "xmlnode";
+tmpDef.args = ["xmlnode parent", "string subnode", "int index"];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Shared;
+DeprecatedDefinitions.push(tmpDef);
+
+// tmpDef = new MTAFunction;
+// tmpDef.label = "onClientChangeNick";
```
Comment (Subtixx) [inline src/defs/deprecated.ts]: Why was this removed?

Comment (Subtixx) [inline src/defs/deprecated.ts:1029]: Why was this removed? https://wiki.multitheftauto.com/wiki/OnClientChangeNick

File: src/defs/deprecated.ts [hunk 2]
```diff
@@ -37,1035 +37,1035 @@ DeprecatedDefinitions.push(tmpDef);
...
+// tmpDef.description = deprecatedText + "Please use ??? ";
+// tmpDef.returnType = "";
+// tmpDef.args = [""];
+// tmpDef.argDescs = {};
+// tmpDef.scriptSide = ScriptSide.Shared;
+// DeprecatedDefinitions.push(tmpDef);
+
+// tmpDef = new MTAFunction;
+// tmpDef.label = "onClientLogout";
```
Comment (Subtixx) [inline src/defs/deprecated.ts:1038]: Why was this removed? https://wiki.multitheftauto.com/wiki/OnClientLogout

**Review by Subtixx — COMMENTED** [review]
Please help me understand why you did what you did ❓ 

Did you also test against duplicates?

EDIT:
Also the whole functions were always scraped using https://github.com/Subtixx/vscode-mtalua/tree/master/tools/wiki_grabber so no worries

File: src/defs/deprecated.ts (hunk 1 above)
Comment (Loki214) [reply src/defs/deprecated.ts:1029]: Well there isn't any support for events so I just commented it for the time being since it would give the wrong format.

**Comment by Loki214** [comment]
What I did was only grab the missing functions that were to the bottom of the files and checked if they were server\client\shared and move them to the right place

File: src/defs/deprecated.ts (hunk 1 above)
Comment (Subtixx) [reply src/defs/deprecated.ts:1029]: That makes sense right. It's an event name not a function. Right thanks for explaining.

**Comment by Subtixx** [comment]
Okay one last question before merging, did you check if there are any duplicates now? By uncommenting this and debugging the extension:
https://github.com/Subtixx/vscode-mtalua/blob/8a18ec8b53c6e9340e695f4bb75913f37f39b4ce/src/features/completionItemProvider.ts#L54-L59

**Comment by Loki214** [comment]
@Subtixx Oh I didn't check for duplicated I'll check now.

EDIT: Done
also when i try debugging the extension I get an error

Activating extension 'subtixx.mtasa-lua' failed: Cannot find module '...vscode-mtalua\out\extension'.

File: src/defs/client.ts [hunk 3]
```diff
@@ -1779,15 +1770,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Client;
 ClientDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "setPlayerHudComponentVisible";
```
Comment (Subtixx) [inline src/defs/client.ts]: This seems to be missing now in your client.ts? setPlayerHudComponentVisible is shared BUT it has different definitions for client / server side.

File: src/defs/server.ts [hunk 4]
```diff
@@ -1256,15 +1237,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Server;
 ServerDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "setPlayerHudComponentVisible";
```
Comment (Subtixx) [inline src/defs/server.ts:1260]: Same here. Since it basically is the same, but has different parameters (which the dupe check doesnt catch) this should be kept.

File: src/defs/server.ts [hunk 5]
```diff
@@ -1954,58 +1926,4 @@ tmpDef.returnType = "bool";
...
-
-tmpDef = new MTAFunction;
-tmpDef.label = "httpWrite";
-tmpDef.description = "This function adds text to the output of the current HTTP file of the HTTP interface. The function can only be used on parsed (i.e not marked as raw) HTTP pages.";
-tmpDef.returnType = "bool";
-tmpDef.args = ["string data [", "int length]"];
-tmpDef.argDescs = {};
-tmpDef.scriptSide = ScriptSide.Server;
-ServerDefinitions.push(tmpDef);
+ServerDefinitions.push(tmpDef);
```
Comment (Subtixx) [inline src/defs/server.ts]: This is now missing a new line ;) Please correct.

File: src/defs/shared.ts [hunk 6]
```diff
@@ -1490,15 +1463,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Shared;
 SharedDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "getElementInterior";
```
Comment (Subtixx) [inline src/defs/shared.ts:1494]: Seems to be missing now?

**Review by Subtixx — COMMENTED** [review]
Haven't checked all of them, but after those who I've highlighted I checked a couple more and except those it seems fine.

File: src/defs/server.ts (hunk 5 above)
Comment (Loki214) [reply src/defs/server.ts]: There needs to be an extra line at the end of the file?
